### ISSUE TITLE
Tutorial should describe the generated test files

### ISSUE THREAD

**Issue by ekuefler** [kubernetes-sigs/kubebuilder#716]
I see that creating an API generated `suite_test.go` in my controller and type directories, and also a test for my types file. Not being very familiar with Ginko, Gomega, or controller-tools I'm not quite sure how I'm intended to write tests, and the tutorial doesn't mention them.

Why does it generate a test file for my type but not my controller? Does that imply I should be creating a test file for the controller myself, or doing everything in `suite_test.go`? What does a reasonable controller test look like? Do I need to create my own mocks for the Client or is one provided somewhere?

**Comment by ekuefler** [comment]
Related: https://github.com/kubernetes-sigs/controller-runtime/blob/master/FAQ.md#q-how-should-i-write-tests--any-suggestions-for-getting-started

**Comment by droot** [comment]
> I'm not quite sure how I'm intended to write tests, and the tutorial doesn't mention them.

Testing is an area where we would like to spend more time to improve the current state. Currently we are just scaffolding the basic skeleton which people can build on. 

> Why does it generate a test file for my type but not my controller?
In previous versions we had CRUD style tests for the types and we realized they really didn't test the user-code much. So we reduced the scaffolding to the minimum with just CREATE test, which is good enough starting point for users to add more tests on the basis of additional validations that users might have added to their types.

About controller tests, we are still figuring out a good sample test for controller that people can use as a reference, so we decided to just provide with the `suite_test.go` to start with. So you are right, the guidance is missing, but we will fill these gaps as we have better idea.
